docs/extending/extending_client_side.md

```diff
@@ -0,0 +1,528 @@
+
+#### [Values](https://stimulus.hotwired.dev/reference/values)
+
+Values are both the initial HTML-driven data for controllers and allow stateful/reactive behaviour. You must declare a value type and an optional default within your controller.
+Values are declared in the initial HTML on the controlled element with the `data-...-value` attribute, for example `<section data-controller="my-controller" data-my-controller-delay-value="300">...</section>`.
+
+#### [Actions](https://stimulus.hotwired.dev/reference/actions)
+
+A simple and powerful way to attach DOM event listeners to any element within your controlled element, letting you react to events such as `click` or `hover` and triggering a Controller method. No JavaScript is needed for this, it's all declarative in your HTML and Stimulus will clean up event listeners for you.
+
+The `data-action` attribute is used, for example `<button type="button" data-action="click->my-controller#open">Open</button>`, where `click` is the event, separated by an `->` and then the controller's identifier with a `#methodName`.
+
+### Adding a custom Stimulus controller
+
+Wagtail exposes a single client-side global (`window.Stimulus`), which is an instance of the core admin Stimulus application.
+
+You need to first create a custom [Stimulus Controller](https://stimulus.hotwired.dev/reference/controllers). This can be done in two main ways, depending on your set-up:
+
+1. Create a **custom Class** that extends the base `window.Stimulus.use.Controller` using class inheritance. If you are using a build tool you can extend `import { Controller } from '@hotwired/stimulus';`.
+2. If you are unable to work with classes, build one with `window.Stimulus.use.createController` which accepts **an object using the [method definitions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Method_definitions) approach**, with a special `STATIC` property for static properties.
+
+Once you have created your custom controller, you will need to [register your Stimulus controllers manually](https://stimulus.hotwired.dev/reference/controllers#registering-controllers-manually) via the `window.Stimulus.register` method.
+
+#### A simple example
+
+First, create your HTML so that appears somewhere within the Wagtail admin.
+
+```html
+<!-- Will log 'My controller has connected: hi' to the console -->
+<div data-controller="my-controller">Hi</div>
+<!-- Will log 'My controller has connected: hello' to the console, with the span element-->
+<div data-controller="my-controller">
+    Hello
+    <span data-my-controller-target="label"></span>
+</div>
+```
+
+Second, create a JavaScript file that will contain your Controller code.
+This Controller logs a simple message on `connect`, which is once the Controller has been created and connected to a HTML element with the matching `data-controller` attribute.
+
+```javascript
+// myapp/static/js/example.js
+
+// 1. With classes
+(() => {
```

Comment:
IMO showing this with a IIFE / proper closure should be out of scope for our docs. All those examples would be clearer without the extra level of indentation and JS idioms.

Reply:
Agreed, removing, may pollute globals if done without the function wrapper but not a problem for simplicity.